modules/renter/contractor/update.go

```diff
@@ -0,0 +1,33 @@
+package contractor
+
+import (
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// renewThreshold calculates an appropriate renewThreshold for a given period.
+func renewThreshold(period types.BlockHeight) types.BlockHeight {
+	threshold := period / 4
+	if threshold > 1000 {
+		threshold = 1000
+	}
+	return threshold
+}
+
+// ProcessConsensusChange will be called by the consensus set every time there
+// is a change in the blockchain. Updates will always be called in order.
+func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.blockHeight != 0 || cc.AppliedBlocks[len(cc.AppliedBlocks)-1].ID() != types.GenesisBlock.ID() {
+		c.blockHeight += types.BlockHeight(len(cc.AppliedBlocks))
+		c.blockHeight -= types.BlockHeight(len(cc.RevertedBlocks))
+	}
```

Comment:
would be even nicer if we had just decided that the genesis block was height 1 :p

I might make sense, except some modules do stuff between updating the height, and some modules don't. We could probably only standardize around the modules that don't change things every block, but I don't think that's very many of them.